### PR TITLE
Persist trainExamples during sessions

### PR DESCRIPTION
**Pull request by evg-tyurin**
* Coach.py holds history of trainExamples from last N iterations as it
stated in AlphaGo paper
* Coach.py is able to save/load examples to/from file

### CODE REVIEW

**Comment by suragnair** [comment]
@evg-tyurin could you remove the "recreate" methods from the NNet files?

**Comment by evg-tyurin** [comment]
Yes, sure.

**Comment by evg-tyurin** [comment]
Done

File: main.py [hunk 1]
```diff
@@ -12,10 +12,14 @@
     'numMCTSSims': 25,
     'arenaCompare': 40,
     'cpuct': 1,
+    'dirAlpha': 0.3,
```
[the remark this answers is not in the document]
Comment (evg-tyurin) [reply main.py]: Removed.

**Comment by evg-tyurin** [comment]
getCheckpointFile - yes.
saveTrainExamples and loadTrainExamples - I think not as "load" method changes the state of Coach (see self.skipFirstSelfPlay).

**Pull request merged**